core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/entity/decorator/PinEntityFilter.java

```diff
@@ -22,7 +21,7 @@ public PinEntityFilter(MemberAccessor memberAccessor) {
     }
 
     @Override
-    public boolean accept(ScoreDirector<Solution_> scoreDirector, Object entity) {
+    public boolean test(Solution_ solution, Object entity) {
```

Comment:
I understand your point about using BiPredicate, but I'm wondering if MovableFilter should have a contract consistent with SelectionFilter. For example, it could have a method named accept instead of test.

Reply:
Also, it seems that accepting the score director instead of the solution class will require fewer changes in other places.

Reply:
The point of this _is_ to accept the solution. Score director is a much heavier class which is not available at all times, and it shouldn't be required when the code doesn't need it.

In fact, this was the original change that triggered the entire PR. For move streams, I need this method to operate without a score director. This change is 100 % intentional.